tests/phpunit/tests/fonts/font-library/wpFontsDir.php

```diff
@@ -69,4 +81,57 @@ function set_new_values( $defaults ) {
 
 		$this->assertSame( static::$dir_defaults, $font_dir, 'The wp_get_font_dir() method should return the default values.' );
 	}
+
+	public function test_should_created_fonts_dir_in_uploads_when_fails_in_wp_content() {
+		// Set the expected results.
+		$upload_dir = wp_upload_dir();
+		$path       = path_join( $upload_dir['basedir'], 'fonts' );
+		$expected   = array(
+			'path'    => $path,
+			'url'     => $upload_dir['baseurl'] . '/fonts',
+			'subdir'  => '',
+			'basedir' => $path,
+			'baseurl' => $upload_dir['baseurl'] . '/fonts',
+			'error'   => false,
+		);
+
+		$this->fake_no_new_directories_in_wp_content();
+		$this->assertFileExists( self::$fake_fonts_file );
+
+		$font_dir = wp_get_font_dir();
+
+		$this->assertDirectoryDoesNotExist( path_join( WP_CONTENT_DIR, 'fonts' ), 'The `wp-content/fonts` directory should not exist' );
+		$this->assertDirectoryExists( $font_dir['path'], 'The `uploads/fonts` directory should exist' );
+		$this->assertSame( $expected, $font_dir, 'The font directory should be a subdir in the uploads directory.' );
+	}
+
+	private function remove_fonts_directory() {
+		$directories = array(
+			path_join( WP_CONTENT_DIR, 'fonts' ),
+			path_join( WP_CONTENT_DIR, 'uploads/fonts' ),
+			'/custom-path/fonts/my-custom-subdir',
+		);
+
+		foreach ( $directories as $dir ) {
+			if ( ! is_dir( $dir ) ) {
+				continue;
+			}
+
+			$this->rmdir( $dir );
+			@rmdir( $dir );
+		}
+	}
+
+	private function fake_no_new_directories_in_wp_content() {
```

Comment:
```suggestion
	/**
	 * A placeholder "fake" file at `wp-content/fonts` triggers `wp_mkdir_p()` to fail into the `file_exists()` bail out, causing `is_dir()` to  return `false`.
	 * This effectively makes `wp-content/fonts` unwritable as the default `fonts` directory location.
	 */
	private function fake_no_new_directories_in_wp_content() {
```
Added a summary of discussion around this approach for future reference (thanks, @hellofromtonya 🙌🏻). It's awkward and could use some massaging if anyone has suggestions! 🙏🏻

Reply:
How about just saying something like this?

```
Creates a file with the same name as the directory. This means the directory cannot be created.
```

That way, it's implementation-independent for possible reuse in the future.